Hazel/src/Platform/Windows/WindowsWindow.cpp

```diff
@@ -172,5 +178,46 @@ namespace Hazel {
 	{
 		return m_Data.VSync;
 	}
+	
+	void WindowsWindow::SetWindowMode(const WindowMode& mode)
+	{
+		HZ_ASSERT(m_Window, "Failed to retrieve window.");
+		if (mode == m_Data.Mode) return;
+
+		unsigned int width = 0, height = 0;
+
+		m_PrimaryMonitor = glfwGetPrimaryMonitor();
+		m_BaseVideoMode = *(glfwGetVideoMode(m_PrimaryMonitor));
+
+		GLFWmonitor* monitor = nullptr;
+
+		if (mode == WindowMode::Windowed) {
+			if (width == 0 || height == 0) {
+				width = m_Data.Width;
+				height = m_Data.Height;
+			}
+			glfwGetWindowPos(m_Window, &(m_Data.XPos), &(m_Data.YPos));
+		}
+		else if (mode == WindowMode::Borderless) {
+			width = m_BaseVideoMode.width;
+			height = m_BaseVideoMode.height;
+			monitor = m_PrimaryMonitor;
+		}
+		else if (mode == WindowMode::FullScreen) {
+			if (width == 0 || height == 0) {
+				// TODO: Change this to check if it is a valid full screen resolution pair
+				width = m_Data.Width;
+				height = m_Data.Height;
+			}
+			monitor = m_PrimaryMonitor;
+		}
+
+		m_Data.Width = width;
+		m_Data.Height = height;
```

Comment:
Here you are setting `m_Data.Width` and `m_Data.Height`.

Assume we are in Borderless mode (now Width & Height equal screen resolution)
Now assume we're switching to windowed mode. Wouldn't windowed mode then show up with same dimensions as Fullscreen, instead of the old Windowed dimensions?

Since we can retreive the Fullscreen/Borderless size anytime, wouldn't it be beter to let m_Data contain the dimensions of the Windowed mode only (same for Xpos and Ypos that are only used for Windowed)?

Reply:
I tested this now and switching to Borderless mode and then back to Windowed mode does indeed put the window size at the larger dimensions from the Borderless Fullscreen.

Reply:
Fixed in https://github.com/TheCherno/Hazel/pull/108/commits/045604ccf8756fc5bde92d93c6d68fd3034debd9